lab_5_scrapper/scrapper.py

```diff
@@ -127,11 +231,31 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        section = article_bs.find_all('section')[1]
+        articles = section.select('a[href^="/2024/"]')
+        url = ''
+        for article in articles:
+            url = 'https://www.fontanka.ru' + article.get('href')
+            if 'all.comments' in url or 'erid' in url:
+                continue
+            if url not in self.urls:
+                break
+        return url
 
     def find_articles(self) -> None:
         """
         Find articles.
         """
+        goal = self.config.get_num_articles()
+        while len(self.urls) < goal:
```

Comment:
are you sure about this condition? is it meaningful to make it a top-level condition?